exporter/loadbalancingexporter/trace_exporter.go

```diff
@@ -90,46 +90,59 @@ func (e *traceExporterImp) Shutdown(context.Context) error {
 func (e *traceExporterImp) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
 	var errs error
 	batches := batchpersignal.SplitTraces(td)
-	for _, batch := range batches {
-		errs = multierr.Append(errs, e.consumeTrace(ctx, batch))
+	endpointToTraceData := make(map[string]ptrace.Traces)
+
+	// Map the trace data to their respective endpoints.
+	for batch := range batches {
+		routingIDs, err := routingIdentifiersFromTraces(batches[batch], e.routingKey)
+		errs = multierr.Append(errs, err)
+		for rid := range routingIDs {
+			endpoint := e.loadBalancer.Endpoint([]byte(rid))
+			if _, ok := endpointToTraceData[endpoint]; ok {
+				// append
+				for i := 0; i < batches[batch].ResourceSpans().Len(); i++ {
+					batches[batch].ResourceSpans().At(i).CopyTo(endpointToTraceData[endpoint].ResourceSpans().AppendEmpty())
+				}
```

Comment:
Why would we check routingIDs has length of 1? Shouldn't we just copy the trace to each backend it's supposed to be routed to, even if that's multiple backends?

Reply:
Well, I mean that in case there is only 1 routing backend - why would we need to copy? In case of >= 2 each backend would share the same traces structure, so we should copy in this case for safety. 

Sure, we can always copy, but there is room for optimization - this is my point

Reply:
I see what you mean. I think we should check for only one endpoint outside of that for loop and routingIDs though. The point of the optimization is that there may be shared endpoints throughout different batches, and different calls to create routingIDs, and those should be sent with the same batched call. I made a function to see if there is just one endpoint there.

Reply:
Good call about this optimization, this could indeed save come RPC calls (not actually what I meant :)
But I have one concern - due to this code we could send one pdata.Traces containing spans with different TraceIDs. This could be a breaking change. 
@jpkrohling Does this exporter guarantee, that in one sent pdata.Traces there is only one TraceID? Or there could be spans from different traces belonging to that backend?

So about my previous point. I meant the situation when there is for example only one trace, which is routed to multiple backends. This way it should be fully copied for each of these backends. And if we see, that there is only one routingID for batch, then I think we could safely move. Maybe we could create a test case for this to be clear what I try to say?

Reply:
>  @jpkrohling Does this exporter guarantee, that in one sent pdata.Traces there is only one TraceID? Or there could be spans from different traces belonging to that backend?

No, there's no such guarantee, we can certainly optimize for that.

Reply:
> I meant the situation when there is for example only one trace, which is routed to multiple backends

Right now, one trace can be sent to only one backend.

Reply:
> Right now, one trace can be sent to only one backend.
Is that so? Why is routingIDs a slice then?

Reply:
One pdata.Traces may contain multiple traces (represented as a ResourceSpan), whose routing keys (service names or trace IDs) are mapped to backends.

Reply:
Returning to my main point - I believe we can move traces only if len(routingIDs) is 1. For simplicity of this PR let's just copy everything as it was in the first place